Source/items.cpp

```diff
@@ -4531,6 +4733,20 @@ void Item::updateRequiredStatsCacheForPlayer(const Player &player)
 	_iStatFlag = player.CanUseItem(*this);
 }
 
+std::string Item::getName() const
+{
+	if (isEmpty()) {

```

Comment:
Let's generate the name once and store it on the item, returning a `string_view` here to avoid doing this on every frame?

Reply:
This has two down-sides
- Make `Item` larger, cause of a new std::string member
- We would need to ensure to invalidate/regenerate the name when the `Item` object changes

To reduce the size impact, we could only store the currently shown name (respect identified flag).
To ensure that the name is correctly invalidated we could introduce a new `Item::initialize` and `Item::identify` method.
The name would only be recreated on-demand (first call to `Item::getName`).

Any thoughts?

Reply:
We could probably also drop the internal/English name and just generate that when saving to save some memory

Reply:
For the a little while I would keep `_iName` and `_iiName`, cause I think a little bit testing of the changes would be good.
Not that there is a bug and the correct names are gone when we are dropping and regenerating the names. 😅 

Also the generation the logic should fast for normal and unidentified items and that's are most of the items on the ground.